### PR TITLE
seqan-apps: fix dependencies and simplify compiler selection

### PR DESCRIPTION
**Pull request by MarcusCalhoun-Lopez**
###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.1
Xcode 11.2 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by rrahn** [comment]
@MarcusCalhoun-Lopez thanks for this contribution. I didn't have time until now to look into it. I see that you removed the requirements for older Xcode versions? I think this reintroduces the error for Xcode 7 which is not fully CXX14 compliant.

**Comment by MarcusCalhoun-Lopez** [comment]
I would be happy to put the code back in.
However, the requirement `compiler.openmp_version 3.0` means that MacPorts Clang (instead of Xcode based Clang) will be used.
Therefore, `minimum_xcodeversions {16 8.1}` would have no effect.
Please let me know if I am missing something.

**Comment by rrahn** [comment]
> However, the requirement compiler.openmp_version 3.0 means that MacPorts Clang (instead of Xcode based Clang) will be used. 

Oh great, that makes sense of course. 

The errror log for Xcode 7.3 says the following  `error: 'shared_timed_mutex' is unavailable: introduced in macOS 10.12`.
So the minimum OS version should be at least macOS 10.12 I guess, but this must be set independent of Xcode then.

**Comment by MarcusCalhoun-Lopez** [comment]
> > However, the requirement compiler.openmp_version 3.0 means that MacPorts Clang (instead of Xcode based Clang) will be used.
> 
> Oh great, that makes sense of course.
> 
> The errror log for Xcode 7.3 says the following `error: 'shared_timed_mutex' is unavailable: introduced in macOS 10.12`.
> So the minimum OS version should be at least macOS 10.12 I guess, but this must be set independent of Xcode then.

Yes, I believe that is correct.
`minimum_xcodeversions {16 8.1}` says on that **on** macOS 10.12, the minimum Xcode required is 8.1.
It does not prevent attempts to build on older systems.

**Comment by rrahn** [comment]
Ok, as far as I could figure it out the minimum os version would be set like this:
```
pre-fetch {
    if { ${os.platform} eq "darwin" && ${os.major} < 16} {
        ui_error "${name} ${version} requires Mac OSX 10.12 or greater."
        return -code error "Incompatible Mac OS X version."
    }
}
```

**Comment by MarcusCalhoun-Lopez** [comment]
I added the code you suggested to fail more gracefully on older OSs.
I also added `known_fail yes`.

File: science/seqan-apps/Portfile [hunk 1]
```diff
@@ -22,7 +20,11 @@ long_description    SeqAn applications for the analysis of large sets of sequenc
 
 homepage            https://www.seqan.de/applications/
 
-depends_build-append    port:boost port:coinor-liblemon
+depends_build-append \
```
Comment (rrahn) [inline science/seqan-apps/Portfile]: Ok, I have double checked and there seems be at least one app that links against boost libraries and not just the headers. So I think boost should be a also a lib dependency.

Comment (MarcusCalhoun-Lopez) [reply science/seqan-apps/Portfile]: The pull request should now have boost as a library dependency.

**Pull request merged**